Modules/posixmodule.c

```diff
@@ -7855,7 +7856,7 @@ os_dup2_impl(PyObject *module, int fd, int fd2, int inheritable)
 
 #endif
 
-    Py_RETURN_NONE;
+    return PyLong_FromLong(res);
```

Comment:
Why don't you rely on Argument Clinic as `os_dup_impl` does?

Reply:
In this case, argument clinic actually complicates the error handling, but I suppose it's the future.